src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AzureServiceBus/AzureServiceBusReceiverManagerWrapper.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Threading.Tasks;
+using NewRelic.Agent.Api;
+using NewRelic.Agent.Extensions.Providers.Wrapper;
+using NewRelic.Reflection;
+
+namespace NewRelic.Providers.Wrapper.AzureServiceBus
+{
+    public class AzureServiceBusReceiverManagerWrapper : AzureServiceBusWrapperBase
+    {
+        private Func<object, object> _receiverAccessor;
+        public override bool IsTransactionRequired => true;
+
+        public override CanWrapResponse CanWrap(InstrumentedMethodInfo instrumentedMethodInfo)
+        {
+            var canWrap = instrumentedMethodInfo.RequestedWrapperName.Equals(nameof(AzureServiceBusReceiverManagerWrapper));
+            return new CanWrapResponse(canWrap);
+        }
+
+        public override AfterWrappedMethodDelegate BeforeWrappedMethod(InstrumentedMethodCall instrumentedMethodCall, IAgent agent, ITransaction transaction)
+        {
+            var receiverManager = instrumentedMethodCall.MethodCall.InvocationTarget;
+            _receiverAccessor ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<object>(receiverManager.GetType(), "Receiver");
```

Comment:
Does `_receiverManager` need to be static (or maybe `static ConcurrentDictionary<type, Func<object, object>>`, since it's possible for `receiverManager` to have multiple types)?